test/tls_test.go

```diff
@@ -0,0 +1,60 @@
+package test
+
+import (
+	"crypto/tls"
+	"testing"
+
+	"github.com/miekg/dns"
+)
+
+func TestDNSoverTLS(t *testing.T) {
+	testcases := []struct {
+		name         string
+		config       string
+		Qname        string
+		Qtype        uint16
+		Answer       []dns.RR
+		AnswerLength int
+	}{
+		{
+			name: "Test whoami with DNS over TLS",
+			config: `tls://.:1053 {
+                tls ../plugin/tls/test_cert.pem ../plugin/tls/test_key.pem
+                whoami
+            }`,
+			Qname:        "example.com.",
+			Qtype:        dns.TypeA,
+			Answer:       []dns.RR{},
+			AnswerLength: 0,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ex, _, tcp, err := CoreDNSServerAndPorts(tc.config)
+			if err != nil {
+				t.Fatalf("Could not get CoreDNS serving instance: %s", err)
+			}
+			defer ex.Stop()
+
+			m := new(dns.Msg)
+			m.SetQuestion(tc.Qname, tc.Qtype)
+			client := dns.Client{
+				Net:       "tcp-tls",
+				TLSConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			r, _, err := client.Exchange(m, tcp)
+
+			if err != nil {
+				t.Fatalf("Could not exchange msg: %s", err)
+			}
+
```

Comment:
can you also check rcode?
```
	if r.Rcode != dns.RcodeSuccess {
		t.Errorf("Expected success but got %d", d.Rcode)
	}
```

Reply:
Sure, added it - and also fixed the golangci-lint